src/engine/Service.cpp

```diff
@@ -359,10 +362,13 @@ TripleComponent Service::bindingToTripleComponent(
   } else if (type == "uri") {
     tc = TripleComponent::Iri::fromIrirefWithoutBrackets(value);
   } else if (type == "bnode") {
-    throw std::runtime_error(
-        "Blank nodes in the result of a SERVICE are currently not supported. "
-        "For now, consider filtering them out using the ISBLANK function or "
-        "converting them via the STR function.");
+    if (!blankNodeMap.contains(value)) {
+      // create a new blankNodeIndex
+      blankNodeMap[value] =
+          Id::makeFromNewBlankNodeIndex(NewBlankNodeIndex::make(
+              getExecutionContext()->getNextNewBlankNodeIndex()));
+    }
```

Comment:
This has to compute the hash of `value` twice or three times in all cases.
Use `try_emplace`. You would have to first insert a dummy Id as the value, but that should not be a problem here.
(see cppreference, ask me if there's trouble).